catroid/src/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -739,6 +745,18 @@ private Object interpretObjectSensor(Sensors sensor, Sprite sprite) {
 			case OBJECT_Y_VELOCITY:
 				returnValue = (double) sprite.look.getYVelocityInUserInterfaceDimensionUnit();
 				break;
+			case OBJECT_LOOK_NUMBER:
+				returnValue = 1.0d + ((lookData != null) ? lookDataList.indexOf(lookData) : 0);
+				break;
+			case OBJECT_LOOK_NAME:
+				returnValue = (lookData != null) ? lookData.getLookName() : null;
+				break;
+			case OBJECT_BACKGROUND_NUMBER:
+				returnValue = 1.0d + ((lookData != null) ? lookDataList.indexOf(lookData) : 0);
+				break;
+			case OBJECT_BACKGROUND_NAME:
+				returnValue = (lookData != null) ? lookData.getLookName() : null;
```

Comment:
`null` is a valid look name. A look can't be named to an empty string `""`.
So I would suggest to return an empty string instead of null. Also children who have never used a programming language have no idea what null means and are definetily confused by it.

If there is no lookdata, look_number will also return 1 which is the same as the first look is set.

Please discuss this with the person responsible for the catrobat language first. Once it's released, it can't be changed.